features/map.go

```diff
@@ -138,6 +136,10 @@ func haveMapType(mt ebpf.MapType) error {
 			err = nil
 		}
 
+	// ENOTSUPP means the map type is at least known to the kernel.
+	case errors.Is(err, unix.ENOTSUPP):
```

Comment:
```suggestion
	case errors.Is(err, unix.ENOTSUPP):
	    if mt == ebpf.StructOpsMap {
	        err = nil
	    }
	    fallthrough // only works if we put it above the err != nil case
```
Should we only accept it for StructOps(Map), but return inconclusive for other (future) map types where it might mean something else? Same for the program probe.

Reply:
Yes, being defensive here might be better.

Reply:
A general observation: `createMapTypeAttr` invites subtle bugs, because we split crafting the syscall payload from checking the specific error we expect. See the calls to `isMapOfMaps`, etc. in `haveMapType`.

Reply:
Sorry, could you explain why this is a problem?